### DISCUSSION TITLE
Multiple fields for one column

### DISCUSSION THREAD

**Discussion by krystof-k** [avo-hq/avo#2737] [Feature feedback]
This is kind of follow up onto #2603. Now I have another issue, consider this:

```ruby
class Avo::Resources::Resource < Avo::BaseResource
  def fields
    field :metadata,
      as: :code,
      language: "json",
      format_using: -> {
        JSON.pretty_generate(value)
      },
      update_using: -> do
        ActiveSupport::JSON.decode(value)
      end

    field :metadata,
      as: :boolean_group,
      name: "Features",
      options: {
        feature_enabled: "Feature enabled",
        another_feature_enabled: "Another feature enabled"
      }
  end
end
```

If I edit the raw JSON content in the first `:code` field, it is always overriden by the `:boolean_group` (and vice versa).

E.g.:

1. Existing metadata content:

   ```json
   {
     "some": "metadata",
     "feature_enabled": false,
     "another_feature_enabled": true
   }
   ```

2. Edited JSON in the `:code` field:

   ```json
   {
     "some": "metadata",
     "more": "metadata",
     "feature_enabled": false,
     "another_feature_enabled": false
   }
   ```

   \+ I tick also _Feature enabled_

3. Result saved:
   ```json
   {
     "some": "metadata",
     "feature_enabled": true,
     "another_feature_enabled": true
   }
   ```

   What I would like/expect:
   ```json
   {
     "some": "metadata",
     "more": "metadata",
     "feature_enabled": true,
     "another_feature_enabled": true
   }
   ```

Is there any approach how to achieve this except creating custom getters/setters? I guess this is caused by updating all values in parallel therefore the second `:boolean_group` field isn't fed with the updated value from the `:code` field – if it would be processing sequentially, it would work.

**Comment by adrianthedev** [comment]
Hmmm, this seems like a pretty unique workflow to edit the same column using two fields.
Also, I think this is more easily fixable using getters and setters rather than trying to "fix" it in Avo.

**Reply by krystof-k** [reply]
Yeah, it probably is. However it may make sense at least throw a warning/error when using two fields for the same column? Which is probably more related to the whole DSL validation thing.

**Reply by adrianthedev** [reply]
I know it sounds trivial to raise that warning, but it's not.
It depends on where the field is being rendered with `hide_on`/`show_on` or even with the `visible` option.
There could be many more edge cases where we have to do crazy gymnastics to figure that out.

I'd rather concentrate on building features that provide value.

**Comment by adrianthedev** [comment]
Closing for now. Feel free to reopen

**Comment by Paul-Bob** [comment]
`for_attribute` might become handy in this scenarios https://docs.avohq.io/3.0/field-options.html#for_attribute

**Reply by krystof-k** [reply]
Yes! That looks like something I may have meant. Will try that out.

**Reply by krystof-k** [reply]
Unfortunately no, it doesn't work at all in this case.

**Reply by Paul-Bob** [reply]
Can you please share your configuration?

**Reply by adrianthedev** [reply]
Or maybe a reproduction repo?

**Reply by krystof-k** [reply]
Well, I was trying something this:

```ruby
    field :metadata,
      as: :code,
      language: "json",
      format_using: -> {
        JSON.pretty_generate(value)
      },
      update_using: -> do
        ActiveSupport::JSON.decode(value)
      end

    field :features,
      as: :boolean_group,
      for_attribute: :metadata,
      options: {
        feature_enabled: "Feature enabled",
        another_feature_enabled: "Another feature enabled"
      }
```

Which ended in this error:

```
ActiveModel::MissingAttributeError in Avo::RoomsController#update
can't write unknown attribute `features`
Extracted source (around line #211):

        def with_value_from_database(value)
          raise ActiveModel::MissingAttributeError, "can't write unknown attribute `#{name}`"
        end
        alias_method :with_value_from_user, :with_value_from_database
        alias_method :with_cast_value, :with_value_from_database

Rails.root: /app

Application Trace | Framework Trace | Full Trace
activemodel (7.0.8.1) lib/active_model/attribute.rb:211:in `with_value_from_database'
activemodel (7.0.8.1) lib/active_model/attribute_set.rb:55:in `write_from_user'
activerecord (7.0.8.1) lib/active_record/attribute_methods/write.rb:36:in `write_attribute'
activerecord (7.0.8.1) lib/active_record/attribute_methods.rb:343:in `[]='
avo (3.7.2) lib/avo/fields/boolean_group_field.rb:38:in `fill_field'
…
```
